src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -851,13 +851,38 @@ impl<'a, 'tcx> PatternContext<'a, 'tcx> {
             ty::TyAdt(adt_def, substs) if adt_def.is_enum() => {
                 match cv.val {
                     ConstVal::Value(val) => {
-                        let discr = const_discr(
+                        let discr_val = const_discr(
```

Comment:
Didn't we discuss that what `const_discr` calls is wrong, and the `Tagged` handling below should be fixed in there too?

Reply:
Ugh the PR just got merged >_<